src/node/channels.h

```diff
@@ -251,6 +248,8 @@ namespace ccf
 
         send_key = nullptr;
         recv_key = nullptr;
+        reset_key_exchange();
+        initiate();
```

Comment:
This is the "hard limit" path. We tried to reset the channel when we hit the soft limit, but if we get spammed with messages ignoring the new key exchange we can end up here. We (correctly) drop the keys, to avoid using them for more encryptions than is safe, but end up in a dead-locked state where the `status` may be in a kind of terminal `ESTABLISHED`, and we'll never try to _re_-establish. So if that happens, reset and re-initiate immediately.